python/tskit/trees.py

```diff
@@ -2910,9 +2929,11 @@ def variants(self, as_bytes=False, samples=None, impute_missing_data=False):
 
         If :ref:`missing data<sec_data_model_missing_data>` is present
         at a given site, the genotypes array will contain a special value
-        ``tskit.MISSING_DATA`` (-1) to identify these missing samples.
-        See the :class:`Variant` class for more details on how missing
-        data is reported.
+        :data:`MISSING_DATA` (-1) to identify these missing samples,
+        and the ``alleles`` tuple will end with the value ``None``
+        (note that this is true whether we specify a fixed mapping
+        using the ``alleles`` parameter or not). See the :class:`Variant`
+        class for more details on how missing data is reported.
```

Comment:
what happens if someone puts `None` in their `alleles` list?

Reply:
They'll get a TypeError. Are you thinking of the case where someone passes in an alleles list that they obtained from another call to ``variants``?

Reply:
I just noticed that missing alleles are None, so it raised the possibility that someone would be confused and think they needed to manually insert it into their alleles list. Raising an error is good.